dotCMS/src/main/java/com/dotmarketing/portlets/templates/action/EditTemplateAction.java

```diff
@@ -319,20 +330,34 @@ else if ((cmd != null) && cmd.equals(com.dotmarketing.util.Constants.FULL_DELETE
 				String [] inodes = req.getParameterValues("publishInode");
 
 				int errorCount =0;
+				SessionDialogMessage errors = new SessionDialogMessage(
+						LanguageUtil.get(user, "Delete-Template"),
+						LanguageUtil.get(user, "message.template.full_delete.error"),
+						LanguageUtil.get(user.getLocale(), "message.template.dependencies.top", 10) + "<br>" +
+								LanguageUtil.get(user.getLocale(), "message.template.dependencies.query",
+										"<br>+baseType:" + BaseContentType.HTMLPAGE.getType()
+								));
+
 				for(String inode  : inodes)	{
 					WebAsset webAsset = (WebAsset) InodeFactory.getInode(inode,Template.class);
 
-					StringBuilder errors = new StringBuilder();
 					if (canTemplateBeDeleted(webAsset, user, errors)) {
 						WebAssetFactory.deleteAsset(webAsset,user);
 					} else {
-						SessionMessages.add(httpReq,SessionMessages.ERROR + errorCount++, errors.toString());
+						if (errorCount == 0) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)